frontend/src/lib/components/Annotations/AnnotationMarker.tsx

```diff
@@ -31,31 +31,31 @@ function coordinateContains(e: MouseEvent, element: DOMRect): boolean {
 }
 
 interface AnnotationMarkerProps {
-    elementId: string
+    elementId?: string
     label: string
-    annotations: AnnotationType[]
+    annotations?: AnnotationType[]
     left: number
     top: number
-    onCreate: (textInput: string, applyAll: boolean) => void
+    onCreate?: (textInput: string, applyAll: boolean) => void
     onDelete?: (annotation: AnnotationType) => void
     onClick?: () => void
     onClose?: () => void
     onCreateAnnotation?: (textInput: string, applyAll: boolean) => void
     size?: number
     color: string | null
     accessoryColor: string | null
-    insightId?: number
-    currentDateMarker: string
+    dashboardItemId?: InsightShortId
```

Comment:
🤔  thanks for the callout. I think I was confused by the different id's here and frankensteined this one. I'll revert this to `insightId: number` so that we can phase out `dashboardItemId`'s from our code.